### PR TITLE
[wpt] Check EdgeDriver version before accepting found binary

### LINKED ISSUE #25416: Edge Dev and Canary broken since September 3rd
There have been no successful Edge Dev or Canary runs since September 3rd ([wpt.fyi runs view](https://wpt.fyi/runs?label=master&label=experimental&max-count=100&product=chrome&product=edge&product=firefox&product=safari&product=webkitgtk)). (In fact, I'm not clear when the last successful Edge Canary run was - can't get the wpt.fyi UI to show me one.

[This seems to be one of the failing runs in Azure Pipelines](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=53958&view=logs&jobId=fb986504-a3b0-5067-98f7-ae723bcc030f&j=2b376673-b6b7-55d3-67e2-e8305abf5c1e). (Safari also fails, which is yet another bug to file...)

From a [random log file](https://dev.azure.com/web-platform-tests/b14026b4-9423-4454-858f-bf76cf6d1faa/_apis/build/builds/53958/logs/178), it looks like there's a mismatch in the version of the edge driver vs the version of edge running:

```
2020-09-06T03:34:58.8271204Z  0:24.86 pid:9376 MSEdgeDriver was started successfully.
2020-09-06T03:34:59.2967628Z  0:25.33 INFO Starting runner
2020-09-06T03:35:02.5877283Z  0:27.81 WARNING Failed to start protocol connection
2020-09-06T03:35:02.5878700Z  0:27.81 WARNING Traceback (most recent call last):
2020-09-06T03:35:02.5879708Z   File "C:\Agent\_work\1\s\tools\wptrunner\wptrunner\executors\protocol.py", line 47, in setup
2020-09-06T03:35:02.5880504Z     self.connect()
2020-09-06T03:35:02.5881265Z   File "C:\Agent\_work\1\s\tools\wptrunner\wptrunner\executors\executorwebdriver.py", line 305, in connect
2020-09-06T03:35:02.5882099Z     self.webdriver.start()
2020-09-06T03:35:02.5882849Z   File "C:\Agent\_work\1\s\tools\webdriver\webdriver\client.py", line 437, in start
2020-09-06T03:35:02.5883672Z     value = self.send_command("POST", "session", body=body)
2020-09-06T03:35:02.5884543Z   File "C:\Agent\_work\1\s\tools\webdriver\webdriver\client.py", line 490, in send_command
2020-09-06T03:35:02.5885283Z     raise err
2020-09-06T03:35:02.5886218Z SessionNotCreatedException: session not created (500): session not created: This version of MSEdgeDriver only supports MSEdge version 84
2020-09-06T03:35:02.5887226Z 
```

Need to check how both are chosen, to see what might be broken.

### PR DESCRIPTION
**Pull request by stephenmcgruer**
Fixes https://github.com/web-platform-tests/wpt/issues/25416

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
This doesn't generalize it to all browsers yet, and I still need to manually verify this works on my Windows machine. (Sadly don't have a MacOS machine to test it, perhaps you could @Hexcles ?)

**Review by Hexcles — COMMENTED** [review]
I got `Unable to get version for EdgeDriver /Users/robertma/github/wpt/_venv2/bin/msedgedriver, rejecting it`

```
$ _venv2/bin/msedgedriver --version
MSEdgeDriver 85.0.564.63 (54dee5e1d729e4c188896f1b1156e98e83557fcc)
```

File: tools/wpt/browser.py [hunk 1]
```diff
@@ -1128,6 +1147,21 @@ def version(self, binary=None, webdriver_binary=None):
...
+    def webdriver_version(self, webdriver_binary):
+        if self.platform == "win":
+            return _get_fileversion(webdriver_binary, self.logger)
+
+        try:
+            version_string = call(webdriver_binary, "--version").strip()
+        except subprocess.CalledProcessError:
+            self.logger.warning("Failed to call %s" % webdriver_binary)
+            return None
+        m = re.match(r"EdgeDriver ([0-9][0-9.]*)", version_string)
```
Comment (Hexcles) [inline tools/wpt/browser.py:1159]:
```suggestion
        m = re.match(r"MSEdgeDriver ([0-9][0-9.]*)", version_string)
```

**Review by Hexcles — COMMENTED** [review]
The version checking and downloading logic seems to work for me on macOS, although I wasn't able to start EdgeDriver Canary:

```
 2:17.69 pid:38222 Full command: /Users/robertma/github/wpt/_venv2/bin/msedgedriver --port=53652 --url-base=/
pid:38222 dyld: Library not loaded: @rpath/libc++.dylib
 2:17.69 pid:38222   Referenced from: /Users/robertma/github/wpt/_venv2/bin/msedgedriver
 2:17.69 pid:38222   Reason: image not found
```

Pretty sure this is a separate issue (EdgeDriver stable seems fine).

File: tools/wpt/browser.py [hunk 2]
```diff
@@ -1128,6 +1147,21 @@ def version(self, binary=None, webdriver_binary=None):
             self.logger.warning("Failed to find Edge binary.")
             return None
 
+    def webdriver_version(self, webdriver_binary):
+        if self.platform == "win":
+            return _get_fileversion(webdriver_binary, self.logger)
```
Comment (Hexcles) [inline tools/wpt/browser.py:1152]: Might want to check this on a Windows machine.

**Review by Hexcles — APPROVED** [review]
LGTM

File: tools/wpt/browser.py (hunk 2 above)
Comment (stephenmcgruer) [reply tools/wpt/browser.py:1152]: Appears to work. Though amusing I found that `channel` isn't obeyed for choosing the Edge binary we grab, so if you have stable installed you always get stable 😆

**Comment by stephenmcgruer** [comment]
This also addresses part of https://github.com/web-platform-tests/wpt/issues/10451

**Pull request merged**